src/celeritas/optical/model/RayleighMfpCalculator.cc

```diff
@@ -0,0 +1,58 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/model/RayleighMfpCalculator.cc
+//---------------------------------------------------------------------------//
+#include "RayleighMfpCalculator.hh"
+
+#include "celeritas/io/ImportOpticalMaterial.hh"
+#include "celeritas/mat/MaterialParams.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct Rayleigh material data from imported data and core Material
+ * parameters.
+ */
+std::vector<OpticalRayleighMaterial> OpticalRayleighMaterial::from_import(
+    std::vector<ImportOpticalRayleigh> const& imported,
+    ::celeritas::MaterialParams const& mat)
```

Comment:
It might make more sense to construct the `RayleighModel` from the imported data and core material params and get rid of the `OpticalRayleighMaterial` altogether.  The mfp calculator could also be constructed with the core `MaterialView` (and the optical properties); then we could just get the temperature from the view and wouldn't have duplicate the other data.

Reply:
My one concern about this was making optical and core physics too interdependent, in case we want to be able to run the optical loop completely independently of the core loop. For example, in running tests we'd have to also pass some mocked version of the core `MaterialParams` to `RayleighModel`.

That being said, what might work best is to instead have 2 constructors for `RayleighModel`, one of which can take the core material params to optionally build missing MFPs, and the other to just require imported MFPs and can run independently of any core data. Thoughts?

Reply:
It does mean if someone wanted to run the optical loop and construct the Rayleigh MFP tables from material data, then they'd also have to initialize the core material params with just the temperatures for each corresponding optical material. Not sure how much of a concern this would be in practice.

Reply:
Yes the material temperature should be part of the core material params—if we don't care about the other materials, and assuming we're not doing doppler broadening for neutrons, we can just set their temperatures to zero.